crates/db_schema/src/impls/tag.rs

```diff
@@ -77,3 +152,49 @@ impl ToSql<Nullable<Json>, Pg> for TagsView {
     <serde_json::Value as ToSql<Json, Pg>>::to_sql(&value, &mut out.reborrow())
   }
 }
+
+impl PostTag {
+  pub async fn update(
+    pool: &mut DbPool<'_>,
+    post: &Post,
+    tag_ids: Vec<TagId>,
+  ) -> LemmyResult<Vec<Self>> {
+    // validate tags
+    let community_tags = Tag::read_for_community(pool, post.community_id)
```

Comment:
Validate before inserting, as its own common function. Never read in an insert or update function.

Reply:
Done